collector.go

```diff
@@ -50,3 +50,71 @@ func (c *Collector[T]) Report(f func(report func(T)) error) Task {
 func (c *Collector[T]) Run(f func() T) Task {
 	return NoError(func() { c.report(f()) })
 }
+
+// A Runner manages a group of [Task] that report values.  At least the Go
+// field must be populated.  A Runner must not be copied after first use.
+type Runner[T any] struct {
+	// Go starts a task in a goroutine. This field must be non-nil.
+	// This field is read-only after initialization.
+	Go func(Task)
+
+	// Collect receives values reported by tasks managed by the Runner.
+	// This field is read-only after initialization.
+	// If nil, values from the tasks are discarded.
+	Collect func(T)
+
+	// Filter filters non-nil errors reported by tasks managed by the Runner.
+	// Its return value replaces the error; if it returns nil, the error is
+	// discarded and the task is treated as successful.
+	// This field is read-only after initialization.
+	// If nil, errors are not filtered.
+	Filter ErrorFunc
+
+	// Hold to call Collect or Filter
+	μ sync.Mutex
+}
+
+// report delivers v to the Collect callback, with the lock held, or if no
+// callback is set it discards v.
+func (r *Runner[T]) report(v T) {
+	if r.Collect == nil {
+		return
+	}
+	r.μ.Lock()
+	defer r.μ.Unlock()
+	r.Collect(v)
+}
+
+// filterErr replaces err (which must be non-nil) with the result of calling
+// the Filter callback with the lock held. If no callback is set, it returns
+// err unmodified.
+func (r *Runner[T]) filterErr(err error) error {
+	if err == nil || r.Filter == nil {
+		return err
+	}
+	r.μ.Lock()
```

Comment:
Hmm, one mutex for errors and values? I think that's a semantic change from the existing collector, but I don't know if I care... I guess this would break if you wanted to wait for an error from the collect func, or vice versa... But that seems a bit far-fetched. Worth a docstring in the struct though, that this pair of optional functions serialize with each other.

Reply:
Ah, good point. I guess it is a semantic change insofar as we've lifted the error filter up one level—previously it's not accessible to the collector, so any error bypasses the collection, whereas here we do that beforehand. (Which is actually what I'd have preferred, though I don't actually have a strong use case for it).

But as-written, you can't wait for an error from the collector anyway—it doesn't report one. (It could, if you think that's useful)